hedera-node/src/main/java/com/hedera/services/state/submerkle/ExpirableTxnRecord.java

```diff
@@ -408,23 +408,30 @@ public void deserialize(SerializableDataInputStream in, int version) throws IOEx
 		}
 	}
 
-	private void deserializeAllowanceMaps(SerializableDataInputStream in) throws IOException {
-		var numCryptoAllowances = in.readInt();
-		if (numCryptoAllowances > 0) {
-			cryptoAllowances = new TreeMap<>();
-		}
-		while (numCryptoAllowances-- > 0) {
-			final EntityNum owner = EntityNum.fromLong(in.readLong());
-			cryptoAllowances.put(owner, deserializeCryptoAllowances(in));
-		}
+	private void deserializeAllowanceMaps(SerializableDataInputStream in, final int version) throws IOException {
+		if (version < RELEASE_0250_VERSION) {
+			// In release 0.24.x three _always-empty_ map sizes were serialized here
```

Comment:
Are these lines correct? The deleted readInt()'s were at the end after the while loop. Notice the original deserializeAllowanceMaps from 0240:

```
	private void deserializeAllowanceMaps(SerializableDataInputStream in) throws IOException {
		var numCryptoAllowances = in.readInt();
		if (numCryptoAllowances > 0) {
			cryptoAllowances = new TreeMap<>();
		}
		while (numCryptoAllowances-- > 0) {
			final EntityNum owner = EntityNum.fromLong(in.readLong());
			cryptoAllowances.put(owner, deserializeCryptoAllowances(in));
		}
		var numTokenAllowances = in.readInt();
		if (numTokenAllowances > 0) {
			fungibleTokenAllowances = new TreeMap<>();
		}
		while (numTokenAllowances-- > 0) {
			final EntityNum owner = EntityNum.fromLong(in.readLong());
			fungibleTokenAllowances.put(owner, deserializeFungibleTokenAllowances(in));
		}

		var numNftAllowances = in.readInt();      // <----- This one got deleted
		if (numNftAllowances > 0) {
			nftAllowances = new TreeMap<>();
		}
		while (numNftAllowances-- > 0) {
			final EntityNum owner = EntityNum.fromLong(in.readLong());    /// <<<--- readLong 
			nftAllowances.put(owner, deserializeNftAllowances(in));            /// <<<--- deserializeNftAllowances
		}
	}
```

Reply:
It's correct; but _only_ because the `CryptoApprove/Allowance` variants were never enabled on a production network. (So for a production state we know each of the three `readInt` calls will return zero.)